src/redux/saga/adItemSaga.ts

```diff
@@ -1,17 +1,18 @@
-import { put } from 'redux-saga/effects';
+import { call, put } from 'redux-saga/effects';
 import { addItemAction } from '../actions/todoActions/addItemAction';
 import { createReferenceHelper } from '../../helpers/createReferenceHelper';
 import { AddItemSagaAction } from '../actions/todoSagaActions/addItemSagaAction';
 import { createAlertMessageHelper } from '../../helpers/createAlertMessageHelper';
 
 export function* addItemSaga(action: AddItemSagaAction) {
-  const { newItem, userToken } = action.payload;
+  const { newItem, userToken, callback } = action.payload;
   try {
     yield createReferenceHelper
       .ref(`Users/${userToken}/Todo/`)
       .child(`${newItem.id}`)
       .set(newItem);
     yield put(addItemAction({ newItem }));
+    yield call(callback);
```

Comment:
what if you catch error? your callback won't be called

Reply:
yes

Reply:
the user should not return to the main screen on error